addons/website_event_track/controllers/event_track.py

```diff
@@ -247,37 +248,37 @@ def _prepare_calendar_values(self, event):
                     'rowspan': duration,  # rowspan
                     'start_date': self._get_locale_time(start_date, lang_code),
                     'end_date': self._get_locale_time(end_date, lang_code),
-                    'occupied_cells': self._get_occupied_cells(track, duration, locations, local_tz)
+                    'occupied_cells': self._get_occupied_cells(track, time_slot, duration, locations, local_tz)
                 }
 
                 # get all the time slots by day to determine the max duration of a day.
                 day = time_slot.date()
                 time_slots_by_day[day]['start'].add(time_slot)
                 time_slots_by_day[day]['end'].add(time_slot+timedelta(minutes=15*duration))
+
+                # count the number of tracks by days
                 tracks_by_days[day] += 1
 
+            # count locations by days
+            for time_slot in time_slots:
+                track_day = time_slot.date()
+                if track.location_id not in locations_by_days[track_day]:
+                    locations_by_days[track_day].append(track.location_id)
+
         # split days into 15 minutes time slots
         global_time_slots_by_day = dict((day, {}) for day in days)
         for day, time_slots in time_slots_by_day.items():
             start_time_slot = min(time_slots['start'])
             end_time_slot = max(time_slots['end'])
```

Comment:
Can't it be just this? So the loop stops at most at the latest possible slot?
```suggestion
            end_time_slot = min(max(time_slots['end']), next_day)
```